receiver/jaegerreceiver/factory.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jaegerreceiver
+
+// This file implements factory for Jaeger receiver.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/census-instrumentation/opencensus-service/consumer"
+	"github.com/census-instrumentation/opencensus-service/internal/configmodels"
+	"github.com/census-instrumentation/opencensus-service/internal/factories"
+	"github.com/census-instrumentation/opencensus-service/receiver"
+)
+
+var _ = factories.RegisterReceiverFactory(&receiverFactory{})
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "jaeger"
+
+	// Protocol values.
+	protoThriftHTTP     = "thrift-http"
+	protoThriftTChannel = "thrift-tchannel"
+
+	// Default endpoints to bind to.
+	defaultHTTPBindEndpoint     = "127.0.0.1:14268"
```

Comment:
hum, I'm not sure that for jaeger these defaults make sense: these are the endpoints for jaeger collector that isn't expected to be on the same network namespace as the clients. The agent in that case should have one of the UDP protocols instead for jaeger (but that is broken at this very moment). I don't think we need to resolve/change this now but c uld you please open an issue to update/change this later?

Reply:
Or perhaps it makes even more sense to separate the jaeger receiver into jaeger-collector and jaeger-agent

Reply:
What is the alternate? "0.0.0.0:14268"? Again, let's keep in mind these are only some sane defaults which will likely be overridden by a default config _file_ that will be included in the distribution. This stays as an open question that I would like to address in due time.

Reply:
"0.0.0.0:14268" will work, but in golang people usually simply write ":14268"